sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -37,24 +44,23 @@
     @Override
     public Flux<FeedResponse<T>> byPage() {
         CosmosPagedFluxOptions cosmosPagedFluxOptions = new CosmosPagedFluxOptions();
-
-        return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
+        return FluxUtil.fluxContext(context ->  byPage(cosmosPagedFluxOptions, context)).subscriberContext(TracerProvider.callDepthAttributeFunc);
```

Comment:
regardless of if tracing is enabled or not we are passing the Pageflux to the tracing stack, and recreating some internal Map in `FluxUtil::toAzureContext`

This should sit behind a tracing enabled flag.

```suggestion
if (tracing.enabled) {
 return FluxUtil.fluxContext(context ->  byPage(cosmosPagedFluxOptions, context)).subscriberContext(TracerProvider.callDepthAttributeFunc);
} else {
 return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
}
```

You should be able to factor out the common if/else code into a method and just call that here, but it seems to me we should have a short circurt for when tracing is not enabled.

Reply:
Cosmos page flux does not hold any state , so we don't have tracing.enabled flag till optionsFluxFunction.apply is called, please suggest

Reply:
done